pkg/store/bucket.go

```diff
@@ -305,6 +346,49 @@ func NewBucketStore(
 	enablePostingsCompression bool,
 	postingOffsetsInMemSampling int,
 	enableSeriesResponseHints bool, // TODO(pracucci) Thanos 0.12 and below doesn't gracefully handle new fields in SeriesResponse. Drop this flag and always enable hints once we can drop backward compatibility.
+) (*BucketStore, error) {
+	bucketStoreOpts := []BucketStoreOption{
+		WithChunksLimit(chunksLimiterFactory),
+		WithChunksSizeLimit(NewChunksLimiterFactory(noChunksSizeLimit)),
+	}
+	return NewBucketStoreWithOptions(logger,
+		reg,
+		bkt,
+		fetcher,
+		dir,
+		indexCache,
+		queryGate,
+		maxChunkPoolBytes,
+		debugLogging,
+		blockSyncConcurrency,
+		filterConfig,
+		enableCompatibilityLabel,
+		enablePostingsCompression,
+		postingOffsetsInMemSampling,
+		enableSeriesResponseHints,
+		bucketStoreOpts...,
+	)
+}
+
+// NewBucketStoreWithOptions creates a new bucket backed store that implements the store API against
+// an object store bucket. It is optimized to work against high latency backends.
+func NewBucketStoreWithOptions(
```

Comment:
@bwplotka What do you think about this? I've mixed feelings about it, just to avoid adding another argument to `NewBucketStore()`. Maybe we can rollback options in this PR and follow up the discussion about functional options in https://github.com/thanos-io/thanos/pull/3931?